reco_utils/evaluation/python_evaluation.py

```diff
@@ -50,32 +46,23 @@ def check_column_dtypes_wrapper(
         *args,
         **kwargs
     ):
-        # check existence of input columns.
-        for col in [col_user, col_item, col_rating]:
-            if col not in rating_true.columns:
-                raise ValueError("schema of y_true not valid. missing {}".format(col))
-
-        for col in [col_user, col_item, col_prediction]:
-            if col not in rating_pred.columns:
-                raise ValueError("schema of y_true not valid. missing {}".format(col))
-
-        # check matching of input column types. the evaluator requires two dataframes have the same
-        # data types of the input columns.
-        if rating_true[col_user].dtypes != rating_pred[col_user].dtypes:
-            raise TypeError(
-                "data types of column {} are different in true and prediction".format(
-                    col_user
-                )
-            )
-
-        if rating_true[col_item].dtypes != rating_pred[col_item].dtypes:
-            raise TypeError(
-                "data types of column {} are different in true and prediction".format(
-                    col_item
-                )
-            )
-
-        return f(
+        """Check columns of DataFrame inputs
+
+        Args:
+            rating_true (pd.DataFrame): True data
+            rating_pred (pd.DataFrame): Predicted data
+            col_user (str): column name for user
+            col_item (str): column name for item
+            col_rating (str): column name for rating
+            col_prediction (str): column name for prediction
+        """
+
+        assert has_columns(rating_true, [col_user, col_item, col_rating])
```

Comment:
I think if we are going to use asserts, we should at least add a message for the user. One question here, why using asserts instead of throwing an exception?

Reply:
FYI @msalvaris

Reply:
It is up to how you think these exceptions will be caught. The best practice is to have a unique exception for different types of errors then functions that try except them can handle them as required. If they are all AssertionError it is going to be hard to do that.

Reply:
Assertions are fine for scripts and notebooks but not ideal for libs

Reply:
asserts are easy, there is already logging inside the function to display an error, but i'm fine putting this back to raising a ValueError.

Reply:
I actually don't agree with the blanket ValueError either tbh. If we are going to use blanket ValueError without any distiction between the errors then asserts are just as good.

Reply:
ahahaha, yes. that was my thinking. but really create a specific exception for this case seems like overkill